hive-metastore/src/main/java/org/apache/iceberg/hive/legacy/MergeHiveSchemaWithAvro.java

```diff
@@ -177,6 +190,15 @@ public Schema listElementPartner(Schema partner) {
       return (schema.getType() == Schema.Type.ARRAY) ? schema.getElementType() : null;
     }
 
+    @Override
+    public Schema unionObjectPartner(Schema partner, int ordinal) {
+      if (partner.getType() != Schema.Type.UNION) {
+        return null;
+      }
+      Schema schema = AvroSchemaUtil.discardNullFromUnionIfExist(partner);
```

Comment:
just curious, why do we need to drop the null? Do they get dropped somewhere else along the way? If so, do we know why & where? Is there e another util to use instead of adding`AvroSchemaUtil.discardNullFromUnionIfExist`?

Reply:
Because in the partner visitor, Hive schema is visited along with the Avro schema, Hive union's schema won't contain null, so we want to discard it temporarily for the benefit of the subsequential nested visiting. Later I added the null back to the avro schema in `AvroSchemaUtil.nullExistInUnion(partner)` condition call.

Reply:
that makes sense. Thanks Raymond!